src/badguy/mriceblock.cpp

```diff
@@ -165,31 +169,28 @@ bool
 MrIceBlock::collision_squished(GameObject& object)
 {
   Player* player = dynamic_cast<Player*>(&object);
+  auto badguy = dynamic_cast<BadGuy*>(&object);
+  auto movingobject = dynamic_cast<MovingObject*>(&object);
```

Comment:
Dynamic casts are extremely expensive and should be avoided if possible, hence why they are declared at the latest moment possible